src/main/java/com/google/api/codegen/util/Name.java

```diff
@@ -192,6 +192,25 @@ public Name join(Name rhs) {
     return new Name(newPieceList);
   }
 
+  public String toOriginal() {
+    if (namePieces.size() != 1) {
+      throw new IllegalArgumentException(
+          "Name: toOriginal can only be called with a namePieces size of 1");
+    }
+
+    CaseFormat caseFormat = namePieces.get(0).caseFormat;
+    switch (namePieces.get(0).caseFormat) {
+      case LOWER_CAMEL:
+      case UPPER_CAMEL:
+        return toCamel(caseFormat);
+      case LOWER_UNDERSCORE:
+      case UPPER_UNDERSCORE:
+        return toUnderscore(caseFormat);
+      default:
+        throw new IllegalArgumentException("Name: CaseFormat.LOWER_HYPHEN not supported.");
+    }
```

Comment:
Woops, forgot to remove this after we decided to only support namePieces of length 1.  Switched.